lib/galaxy/tool_util/verify/asserts/tabular.py

```diff
@@ -16,4 +19,4 @@ def assert_has_n_columns(output, n, sep='\t'):
     n = int(n)
     first_line = get_first_line(output)
     assert first_line is not None, "Was expecting output with %d columns, but output was empty." % n
-    assert len(first_line.split(sep)) == n, "Output does not have %d columns." % n
+    assert len(first_line.split(sep)) == n, f"Expected {n} columns in output, found {first_line.split(sep)} lines"
```

Comment:
```suggestion
    n_columns = len(first_line.split(sep))
    assert n_columns == n, f"Expected {n} columns in output, found {n_columns} columns"
```

Can we add a parameter here that controls whether we skip non-comment and non-empty lines (and what the comment character is) ? Alternatively, can we get the number of columns form the dataset metadata ?

Reply:
> Alternatively, can we get the number of columns form the dataset metadata ?

This would be nice, but I think the functions get plain text file content.

> Can we add a parameter here that controls whether we skip non-comment and non-empty lines (and what the comment character is) ?

How about adding a `comment` attribute which is empty by default. If given lines starting with any of the characters are ignored.